### ISSUE TITLE
"Literal['A'] cannot be assigned to parameter" for generic bound by `Literal['A']`

### ISSUE THREAD

**Issue by lmmarsano** [microsoft/pyright#1667]
**Describe the bug**
Type error for passing literal value through generically typed parameter bound by a compatible literal type.

**To Reproduce**
Save the provided code as `stub.py` & run the CLI.

**Expected behavior**
No type error.

**Screenshots or Code**
```python
import typing as T
_V = T.TypeVar('_V', bound=T.Literal['A'])
def id(x: _V) -> _V:
	return x
id('A')
```

**VS Code extension or command-line**
```shellsession
$ npx pyright stub.py --outputjson
```
```json
{
    "version": "1.1.123",
    "time": "1616275476457",
    "diagnostics": [
        {
            "file": "/home/LOM0227/project/ALM.OnRampSecrets/stub.py",
            "severity": "error",
            "message": "Argument of type \"Literal['A']\" cannot be assigned to parameter \"x\" of type \"_V@id\" in function \"id\"\n  Type \"str\" is incompatible with bound type \"Literal['A']\" for type variable \"_V@id\"\n    \"str\" cannot be assigned to type \"Literal['A']\"",
            "range": {
                "start": {
                    "line": 4,
                    "character": 3
                },
                "end": {
                    "line": 4,
                    "character": 6
                }
            },
            "rule": "reportGeneralTypeIssues"
        }
    ],
    "summary": {
        "filesAnalyzed": 14,
        "errorCount": 1,
        "warningCount": 0,
        "informationCount": 0,
        "timeInSec": 0.259
    }
}
```

**Comment by erictraut** [comment]
Out of curiosity, where would you ever use a literal bound like this? There's no way to further narrow the type of `Literal["A"]`, so the only possible type that could solve the constraint is `Literal["A"]`. And if that's the case, why use a type variable?

**Comment by erictraut** [comment]
I'm still not sure where this would ever be useful in practice, but I've fixed the underlying cause of the false positive.

This will be fixed in the next release of pyright.

**Comment by lmmarsano** [comment]
@erictraut, simplifications/reductions can look impractical.

The motivation had a signature like
```python
_CM = T.TypeVar('_CM', bound=T.Literal['Automatic', 'Manual'])
"""Creation Mode"""

def fun(parameter: _CM) -> Structure[_CM]: ...
```
where `fun('Automatic')` would preferably resolve to `Structure[T.Literal['Automatic']]`.

**Comment by mozharovsky** [comment]
This issue also affected configs with dataclasses:

```python
Format = Literal["json", "jsonl", "csv", "text"]

@dataclass
class ReaderConfig:
    format: Format = field(
        default="text",
        metadata={"help": "The format of reading documents."},
    )
```

**Comment by erictraut** [comment]
@mozharovsky, that's a different case. It doesn't involve a bound TypeVar, so it's not affected by this change.

**Comment by erictraut** [comment]
I've filed a separate bug to track the other issue: https://github.com/microsoft/pyright/issues/1673

**Comment by mozharovsky** [comment]
> @mozharovsky, that's a different case. It doesn't involve a bound TypeVar, so it's not affected by this change.

Oh, I see what you mean, my case doesn't involve bounded TypeVar. Thanks for making this clear!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.124, which I just published. It will also be included in the next release of pylance.
